api/internalutils/stream/stream_test.go

```diff
@@ -348,3 +363,185 @@ func TestCollectPages(t *testing.T) {
 		})
 	}
 }
+
+// TestRateLimitFailure verifies the expected failure conditions of the RateLimit helper.
+func TestRateLimitFailure(t *testing.T) {
+	t.Parallel()
+
+	var limiterError = fmt.Errorf("limiter-error")
+	var streamError = fmt.Errorf("stream-error")
```

Comment:
Nit: use `errors.New("limiter-error")` since you don't have any format verbs that require `Errorf`.